### PR TITLE
Py ak create restaurant branch

### PR DESCRIPTION
**Pull request by APKirstein**
Added feature to create new restaurants

### CODE REVIEW

File: spec/features/restaurant/user_sees_all_restaurants_spec.rb [hunk 1]
```diff
@@ -10,17 +10,17 @@
...
-    expect(page).to have_content(restaurant.state)
   end
 
-  scenario 'user sees restaurants ordered by ranking' do
-    skip
-  end
+  # restaurant_2 = FactoryGirl.create(:restaurant)
+  # scenario 'user sees restaurants ordered by ranking' do
+  #   visit '/'
+  #   page.body.index(restaurant.name).should < page.body.index(restaurant_2.name)
```
Comment (APKirstein) [inline spec/features/restaurant/user_sees_all_restaurants_spec.rb]: Soft wrap issues

File: app/views/restaurants/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<ul>
+  <% @restaurants.each do |restaurant| %>
+    <li><%= link_to "#{restaurant.name}", restaurant_path(restaurant) %></li>
+  <% end %>
+</ul>
+
+<h3><center><%= link_to "New", new_restaurant_path %></center></h3>
```
Comment (cmkoller) [inline app/views/restaurants/index.html.erb]: Don't use the html element to center here! Instead, if you want to center this text, add the class `text-center` directly to the `h3`. (This class is provided by Foundation.) It's better form to keep non-content-based stuff out of your html tags

File: spec/features/restaurant/user_creates_new_restaurant_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  As a user
+  I want to visit the new restaurant page
+  so I can add a new restaurant
+} do
+
+  # [ ] - I can navigate to the restaurants/new page
+  # [ ] - I can create a new restaurant
+  # [ ] - I am redirected to the restaurants index page
+
+  restaurant = FactoryGirl.build(:restaurant)
```
Comment (cmkoller) [inline spec/features/restaurant/user_creates_new_restaurant_spec.rb]: You shouldn't be using normal variable assignment outside of your `scenario`s. I'd recommend using a `let` statement here instead.

File: spec/features/restaurant/user_sees_all_restaurants_spec.rb [hunk 4]
```diff
@@ -10,17 +10,17 @@
   # [ ] - I can view a list of restaurants
   # [ ] - I can view restaurnts in ranked order
 
-  let(:restaurant) { Restaurant.create!(name: "Chipotle", address: "33 Harrison Ave", city: "Boston", state: "MA") }
+  restaurant = FactoryGirl.create(:restaurant)
```
Comment (cmkoller) [inline spec/features/restaurant/user_sees_all_restaurants_spec.rb]: Make this a `let` as well

File: spec/features/restaurant/user_creates_new_restaurant_spec.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    fill_in('State', with: restaurant.state)
+    fill_in('Zip', with: restaurant.zip_code)
+
+    click_button('Add Restaurant')
+
+    expect(page).to have_content(restaurant.name)
+
+  end
+
+  #invalid test
```
Comment (PVUL) [inline spec/features/restaurant/user_creates_new_restaurant_spec.rb]: Fixed

File: spec/features/restaurant/user_creates_new_restaurant_spec.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  so I can add a new restaurant
+} do
+
+  # [ ] - I can navigate to the restaurants/new page
+  # [ ] - I can create a new restaurant
+  # [ ] - I am redirected to the restaurants index page
+
+  restaurant = FactoryGirl.build(:restaurant)
+
+  #valid test
```
Comment (PVUL) [inline spec/features/restaurant/user_creates_new_restaurant_spec.rb]: Fixed

File: app/controllers/restaurants_controller.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  def index
+    @restaurants = Restaurant.all
+  end
+
+  def show
+    @restaurant = Restaurant.find(params[:id])
+  end
+
+  def new
+   @restaurant = Restaurant.new
```
Comment (PVUL) [inline app/controllers/restaurants_controller.rb]: Fixed

File: app/views/restaurants/index.html.erb (hunk 2 above)
Comment (PVUL) [inline app/views/restaurants/index.html.erb]: Fixed.

File: spec/features/restaurant/user_creates_new_restaurant_spec.rb (hunk 3 above)
Comment (PVUL) [inline spec/features/restaurant/user_creates_new_restaurant_spec.rb]: Fixed.

File: spec/features/restaurant/user_sees_all_restaurants_spec.rb (hunk 4 above)
Comment (PVUL) [inline spec/features/restaurant/user_sees_all_restaurants_spec.rb]: Fixed.

**Pull request merged**